src/watertap_contrib/reflo/analysis/example_flowsheets/test/test_fo_trevi_flowsheet.py

```diff
@@ -0,0 +1,132 @@
+)
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+)
+from watertap_contrib.reflo.analysis.example_flowsheets.fo_trevi_flowsheet import (
+    build_fo_trevi_flowsheet,
+    fix_dof_and_initialize,
+    get_flowsheet_performance,
+)
+from watertap_contrib.reflo.costing import TreatmentCosting
+from watertap.core.solvers import get_solver
+
+solver = get_solver()
+
+
+class TestTreviFO:
+    @pytest.fixture(scope="class")
+    def fo_trevi_frame(self):
+
+        m = build_fo_trevi_flowsheet(
+            recovery_ratio=0.3,  # Assumed FO recovery ratio
+            RO_recovery_ratio=0.9,  # RO recovery ratio
+            NF_recovery_ratio=0.8,  # Nanofiltration recovery ratio
+            dp_brine=0,  # Required pressure over brine osmotic pressure (Pa)
+            heat_mixing=105,  # Heat of mixing in the membrane (MJ/m3 product)
+            separation_temp=90,  # Separation temperature of the draw solution (C)
+            separator_temp_loss=1,  # Temperature loss in the separator (K)
+            feed_temperature=13,  # Feed water temperature (C)
+            feed_vol_flow=3.704,  # Feed water volumetric flow rate (m3/s)
+            feed_TDS_mass=0.035,  # TDS mass fraction of feed
+            strong_draw_temp=20,  # Strong draw solution inlet temperature (C)
+            strong_draw_mass=0.8,  # Strong draw solution mass fraction
+            product_draw_mass=0.01,  # Mass fraction of draw in the product water
+        )
+
+        fix_dof_and_initialize(
+            m,
+            strong_draw_mass_frac=0.8,
+            product_draw_mass_frac=0.01,
+            RO_recovery_ratio=0.9,
+            NF_recovery_ratio=0.8,
+        )  # same input as above
+
+        # Specify the temperature of the weak draw solution and product water after going through HX1
+        m.fs.HX1.overall_heat_transfer_coefficient[0].unfix()
+        m.fs.HX2.overall_heat_transfer_coefficient[0].unfix()
+        m.fs.HX1.weak_draw_outlet.temperature.fix(80 + 273.15)
+        m.fs.HX1.product_water_outlet.temperature.fix(28 + 273.15)
+
+        return m
+
+    @pytest.mark.unit
+    def test_dof(self, fo_trevi_frame):
+        m = fo_trevi_frame
+        assert degrees_of_freedom(m) == 0
+
+    @pytest.mark.component
+    def test_solve(self, fo_trevi_frame):
+        m = fo_trevi_frame
+
+        results = solver.solve(m)
+        assert_optimal_termination(results)
+
+    @pytest.mark.component
+    def test_costing(self, fo_trevi_frame):
+        m = fo_trevi_frame
+        # Add cost package of Trevi FO system
+        m.fs.costing = TreatmentCosting()
+        m.fs.costing.base_currency = pyunits.USD_2020
+
+        # Create cost block for FO
+        m.fs.fo.costing = UnitModelCostingBlock(flowsheet_costing_block=m.fs.costing)
+
+        # Add LCOW component
+        m.fs.costing.cost_process()
+        m.fs.costing.add_annual_water_production(m.fs.system_capacity)
+        m.fs.costing.add_LCOW(m.fs.system_capacity)
```

Comment:
Can you add a check that ```m.fs.system_capacity``` equals however this flow is being represented in the unit model (i.e. ```product_prop.flow_vol_phase["Liq"]``` or whatever.)

Reply:
Sure, check is now added.